### PR TITLE
Add support for future versions of dbt-core

### PR DESCRIPTION
**Pull request by smith-m**
## Overview

- Description: Unpin dbt-core to allow library users to adopt newer versions of dbt-core 
- Related issue(s): https://github.com/starburstdata/dbt-trino/issues/55
- Related link(s): https://docs.getdbt.com/docs/core-versions#upgrading-to-new-patch-versions

## Checklist

- [x] This PR includes tests, or tests are not required/relevant for this PR - not relevant
- [x] `README.md` updated and added information about my change - no change for users
- [x] `CHANGELOG.md` updated and added information about my change

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def _dbt_trino_version():
         ]
     },
     install_requires=[
-        "dbt-core=={}".format(dbt_version),
```
Comment (hovaesco) [inline setup.py:62]: What about `dbt-core~={}` ?

**Comment by hovaesco** [comment]
Please squash Update Changelog commits and capitalize both.

File: setup.py (hunk 1 above)
Comment (jtcohen6) [reply setup.py:62]: Yes! We recommend using `~=` (compatibility operator)

**Comment by hovaesco** [comment]
Resolved by https://github.com/starburstdata/dbt-trino/commit/6822010551b647d92e514ea60d21b3023af888ed

**Pull request closed without merging**